src/main/java/teammates/common/datatransfer/StudentProfileAttributes.java

```diff
@@ -65,90 +66,97 @@ public StudentProfileAttributes() {
         this.pictureKey = "";
         this.modifiedDate = null;
     }
-    
+
     public String generateUpdateMessageForStudent() {
-        String reminder = "";
-        
         if (isMultipleFieldsEmpty()) {
-            reminder = Const.StatusMessages.STUDENT_UPDATE_PROFILE; 
+            return Const.StatusMessages.STUDENT_UPDATE_PROFILE;
         } else {
             if (this.shortName.isEmpty()) {
-                reminder = Const.StatusMessages.STUDENT_UPDATE_PROFILE_SHORTNAME; 
+                return Const.StatusMessages.STUDENT_UPDATE_PROFILE_SHORTNAME;
             } else if (this.email.isEmpty()) {
-                reminder = Const.StatusMessages.STUDENT_UPDATE_PROFILE_EMAIL; 
+                return Const.StatusMessages.STUDENT_UPDATE_PROFILE_EMAIL;
             } else if (this.pictureKey.isEmpty()) {
-                reminder = Const.StatusMessages.STUDENT_UPDATE_PROFILE_PICTURE;
+                return Const.StatusMessages.STUDENT_UPDATE_PROFILE_PICTURE;
             } else if (this.moreInfo.isEmpty()) {
-                reminder = Const.StatusMessages.STUDENT_UPDATE_PROFILE_MOREINFO;
+                return Const.StatusMessages.STUDENT_UPDATE_PROFILE_MOREINFO;
             } else if (this.nationality.isEmpty()) {
-                reminder = Const.StatusMessages.STUDENT_UPDATE_PROFILE_NATIONALITY;
+                return Const.StatusMessages.STUDENT_UPDATE_PROFILE_NATIONALITY;
             }
+            return "";
         }
-        
-        return reminder;
     }
-    
+
     private boolean isMultipleFieldsEmpty() {
-        int numEmptyFields = 0;
-        if (this.shortName.isEmpty()) {
-            numEmptyFields ++;
-        }
-        if (this.email.isEmpty()) {
-            numEmptyFields ++;
-        }
-        if (this.nationality.isEmpty()) {
-            numEmptyFields ++;
-        }
-        if (this.moreInfo.isEmpty()) {
-            numEmptyFields ++;
-        }
-        if (this.pictureKey.isEmpty()) {
-            numEmptyFields ++;
-        }
-        
+        int numEmptyFields = returnOneIfEmpty(shortName)
+                + returnOneIfEmpty(email)
+                + returnOneIfEmpty(nationality)
+                + returnOneIfEmpty(moreInfo)
+                + returnOneIfEmpty(pictureKey);
         return numEmptyFields > 1;
     }
 
+    private int returnOneIfEmpty(String str) {
+        return str.isEmpty() ? 1 : 0;
+    }
```

Comment:
Good job on naming the method based on what it does. In addition, try to pick a name that reads well in the context where it is used. The word `return` does not fit well into places where the method is used. May be you need a method called `countEmptyStrings(String... strings)`? The `...` means the method can take a variable number of parameters. Your usage context will look something like `numEmptyFields = countEmptyStrings(shortName, email, moreInfo)` The method can be in the `StringHelper` class.